modules/compatibility/src/test/java/org/apache/ignite/compatibility/sql/randomsql/RandomQuerySupplier.java

```diff
@@ -202,6 +205,15 @@ private Select rndSingleTableSelect(RandomisedQueryContext rndQryCtx) {
     private Ast rndQueryFilter(RandomisedQueryContext rndQryCtx) {
         List<TableRef> scopeTbls = rndQryCtx.scopeTables();
 
+        RandomisedQueryContext parentCtx = rndQryCtx.parentContext();
+        if (parentCtx != null) {
```

Comment:
Should we have some documentation about what filters can be generated with this tools? Because now it's become not obvious what conditions can be generated by this tool.

Reply:
What kind of documentation do you mean?

Reply:
Maybe just javadoc with all types of queries or some other logic description. For example - why we must generate some special conditions for subquery?